### PR TITLE
Touch conditions now handle mouse too.

### PR DESCRIPTION
**Pull request by D8H**
It allows to only handle multi-touch and have a basic handling of the mouse automatically.

### Changes
- When the mouse left button is pressed, it's seen as a touch with touchId = 1.
- Touches use id from 2 to reserve 1 for the mouse.
- The compatibility with old events is covered with tests.
- The draggable behavior is simplified.

### Manual tests
- Tested on the current button extension
- Tested on the button update.
- Tested on the draggable behavior.

### CODE REVIEW

**Review by 4ian — COMMENTED** [review]
First quick pass

File: GDJS/Runtime/inputmanager.ts [hunk 1]
```diff
@@ -26,11 +27,26 @@ namespace gdjs {
     _lastPressedKey: float = 0;
     _pressedMouseButtons: Array<boolean>;
     _releasedMouseButtons: Array<boolean>;
+    _cursorX: float = 0;
+    _cursorY: float = 0;
     _mouseX: float = 0;
     _mouseY: float = 0;
```
Comment (4ian) [inline GDJS/Runtime/inputmanager.ts:33]: I think we should comment why we both have mouseX/Y and cursorX/Y

File: GDJS/Runtime/inputmanager.ts [hunk 2]
```diff
@@ -181,8 +209,41 @@ namespace gdjs {
...
+      if (this.isMouseButtonPressed(InputManager.MOUSE_LEFT_BUTTON)) {
+        this._moveTouch(
+          InputManager.MOUSE_TOUCH_ID,
+          this.getCursorX(),
+          this.getCursorY()
+        );
+      }
+    }
+
+    _setMousePosition(x: float, y: float): void {
```
Comment (4ian) [inline GDJS/Runtime/inputmanager.ts:224]: Nit: Should this be `_setCursorPosition`?

File: Core/GDCore/Extensions/Builtin/MouseExtension.cpp [hunk 3]
```diff
@@ -361,13 +403,50 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsMouseExtension(
...
+          "res/conditions/touch.png")
+      .AddCodeOnlyParameter("currentScene", "")
       .AddParameter("expression", _("Touch index"));
 
   extension
       .AddCondition(
           "HasTouchEnded",
           _("A touch has ended"),
-          _("Check if a touch has ended."),
+          _("Check if a touch has ended or a mouse left button has been released."),
```
Comment (4ian) [inline Core/GDCore/Extensions/Builtin/MouseExtension.cpp:449]: This is a change to an existing condition.
Should this be SetHidden and a new condition `HasTouchIrMouseEnded` - like the previous?

Comment (D8H) [reply Core/GDCore/Extensions/Builtin/MouseExtension.cpp:449]: The touch id for the mouse is given at the start, there is no reason to check the mouse id at the end as the legacy expressions wont give it.

**Comment by 4ian** [comment]
LGTM

**Pull request merged**